numbat/src/command.rs

```diff
@@ -398,91 +655,181 @@ mod test {
+            "list units",
+            BareCommand::List {
+                items: Some(ListItems::Units),
+            },
+        );
 
-        assert!(parse!("info").is_err());
-        assert_eq!(parse!("info arg").unwrap(), Command::Info { item: "arg" });
-        assert_eq!(parse!("info .").unwrap(), Command::Info { item: "." });
-        assert!(parse!("info arg1 arg2").is_err());
+        expect_ok(&runner, &mut ctx, "quit", BareCommand::Quit);
+        expect_fail(&runner, &mut ctx, "quit arg");
+        expect_fail(&runner, &mut ctx, "quit arg1 arg2");
 
-        assert_eq!(parse!("clear").unwrap(), Command::Clear);
-        assert!(parse!("clear arg").is_err());
-        assert!(parse!("clear arg1 arg2").is_err());
+        expect_ok(&runner, &mut ctx, "exit", BareCommand::Quit);
+        expect_fail(&runner, &mut ctx, "exit arg");
+        expect_fail(&runner, &mut ctx, "exit arg1 arg2");
 
-        assert_eq!(parse!("list").unwrap(), Command::List { items: None });
-        assert_eq!(
-            parse!("list functions").unwrap(),
-            Command::List {
-                items: Some(ListItems::Functions)
-            }
+        expect_ok(
+            &runner,
+            &mut ctx,
+            "save",
+            BareCommand::Save { dst: "history.nbt" },
         );
-        assert_eq!(
-            parse!("list dimensions").unwrap(),
-            Command::List {
-                items: Some(ListItems::Dimensions)
-            }
-        );
-        assert_eq!(
-            parse!("list variables").unwrap(),
-            Command::List {
-                items: Some(ListItems::Variables)
-            }
-        );
-        assert_eq!(
-            parse!("list units").unwrap(),
-            Command::List {
-                items: Some(ListItems::Units)
-            }
+        expect_ok(
+            &runner,
+            &mut ctx,
+            "save arg",
+            BareCommand::Save { dst: "arg" },
         );
+        expect_ok(&runner, &mut ctx, "save .", BareCommand::Save { dst: "." });
+        expect_fail(&runner, &mut ctx, "save arg1 arg2");
+    }
+
+    #[test]
+    fn test_runner() {
+        fn test_case(
+            runner: &CommandRunner<()>,
+            ctx: &mut Context,
+            input: &'static str,
+            expected: CommandControlFlow,
+        ) {
+            let args = CommandContext {
+                ctx,
+                editor: &mut (),
+            };
+            assert_eq!(expected, runner.try_run_command(input, args).unwrap());
+        }
 
-        assert_eq!(parse!("quit").unwrap(), Command::Quit);
-        assert!(parse!("quit arg").is_err());
-        assert!(parse!("quit arg1 arg2").is_err());
+        let mut ctx = Context::new_without_importer();
 
-        assert_eq!(parse!("exit").unwrap(), Command::Quit);
-        assert!(parse!("exit arg").is_err());
-        assert!(parse!("exit arg1 arg2").is_err());
+        let runner = CommandRunner::<()>::new_all_disabled()
```

Comment:
This would be
```suggestion
        let runner = CommandRunner::new_all_disabled()
```
with the suggestion above.